client/app/components/tags-control/TagsControl.jsx

```diff
@@ -18,87 +21,118 @@ export default class TagsControl extends React.Component {
     getAvailableTags: () => Promise.resolve([]),
     onEdit: () => {},
     className: '',
+    children: null,
   };
 
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      showModal: false,
-    };
-
-    // get available tags
-    this.props.getAvailableTags()
-      .then((tags) => {
-        this.availableTags = tags;
-      });
+  static Prepend({ children }) {
+    return <React.Fragment>{children}</React.Fragment>;
   }
 
-  onTagsChanged = (newTags) => {
-    this.props.onEdit(newTags);
-    this.closeEditModal();
+  static Append({ children }) {
+    return <React.Fragment>{children}</React.Fragment>;
   }
 
+  state = {
+    showModal: false,
+    availableTags: [],
+  };
+
   openEditModal = () => {
-    this.setState({ showModal: true });
-  }
+    // load available tags every time modal is shown; show modal only when tags loaded
+    this.props.getAvailableTags()
+      .then(availableTags => this.setState({ showModal: true, availableTags }));
```

Comment:
Not sure this is the best approach.
It might mean that there would be a noticeable (even if slight) irresponsiveness to the click event.

Perhaps have the modal open but the `<Select>` disabled and indicating loading until promise resolved.
```js
openEditModal = () => {
  this.setState({ showModal: true, loading: true });
  this.props.getAvailableTags()
      .then(availableTags => this.setState({ loading: false, availableTags }));
};

<Select disabled={this.state.loading} loading={this.state.loading} >
  {this.selectOptions}
</Select>
```

Reply:
Then I should move this to modal. But not a big deal